### PR TITLE
Update to Babel 7

### PR DESCRIPTION
**Pull request by kibertoad**
implements #791

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -31,7 +31,7 @@ rules:
 
   # Best Practices
   # http://eslint.org/docs/rules/#best-practices
-  no-multi-spaces: 2
+  no-multi-spaces: 0
```
Comment (kibertoad) [inline .eslintrc]: This was actually failing linting, hence I disabled it

Comment (arthurschreiber) [reply .eslintrc]: Mmh... how exactly was this rule failing the linting? 🤔 I’d check myself, but until the end of this week I only have access to an iPad and no real computer to do so. 🙇🏻‍♂️

Comment (kibertoad) [reply .eslintrc]: @arthurschreiber 

```
/Users/igor.savin/WebstormProjects/tedious-f2/src/bulk-load.js
  15:29  error  Multiple spaces found before '// introduced ...'  no-multi-spaces
  16:29  error  Multiple spaces found before '// introduced ...'  no-multi-spaces
  18:29  error  Multiple spaces found before '// introduced ...'  no-multi-spaces
  19:29  error  Multiple spaces found before '// introduced ...'  no-multi-spaces
  20:29  error  Multiple spaces found before '// introduced ...'  no-multi-spaces

/Users/igor.savin/WebstormProjects/tedious-f2/src/connection.js
   765:60  error  Multiple spaces found before '// EOM pseudo ...'  no-multi-spaces
   857:60  error  Multiple spaces found before '// release old...'  no-multi-spaces
  1475:60  error  Multiple spaces found before '// Request.pau...'  no-multi-spaces
  1759:60  error  Multiple spaces found before '// request tim...'  no-multi-spaces

/Users/igor.savin/WebstormProjects/tedious-f2/src/token/stream-parser.js
  38:60  error  Multiple spaces found before '// generate en...'  no-multi-spaces

/Users/igor.savin/WebstormProjects/tedious-f2/test/unit/login7-payload-test.js
  100:53  error  Multiple spaces found before '// Length'      no-multi-spaces
  101:53  error  Multiple spaces found before '// Variable'    no-multi-spaces
  107:53  error  Multiple spaces found before '// Reserved'    no-multi-spaces
  112:47  error  Multiple spaces found before '// NTLM'        no-multi-spaces
  115:53  error  Multiple spaces found before '// cbSSPILong'  no-multi-spaces
```

**Comment by kibertoad** [comment]
@arthurschreiber Is linting failure a blocker? Do I need to reassemble PR again?

**Comment by arthurschreiber** [comment]
`commitlint` failures are not necessary a blocker, but I’d like to see the ESLint failures fixed via a separate PR so that the changes in here can be focused on just the Babel upgrade. As mentioned before, I’m traveling right now until the end of the week and can not really perform the changes myself. 😔

**Comment by kibertoad** [comment]
@arthurschreiber They are kinda related, newer version of babel-eslint requires newer version of eslint that fails with current codebase. Do you prefer to have separate PRs where the first one would result in broken state?..

**Comment by arthurschreiber** [comment]
No, I think the offending comments can be changed and made to pass in the version of ESLint used on `master`. Their indentation is obviously wrong, but this was not correctly picked up by earlier ESLint versions. By changing these comments via a separate PR that can be merged first, ESLint can be upgraded in this PR without having to touch the ESLint config at all. 😊

**Comment by kibertoad** [comment]
@arthurschreiber Gotcha! Will fix within an hour.

**Comment by kibertoad** [comment]
@arthurschreiber Merged with master, restored original ESLint rules.

**Pull request merged**